variable/sort.cpp

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Thibault Chatel
+#include <numeric>
+
+#include "scipp/variable/sort.h"
+#include "scipp/variable/subspan_view.h"
+#include "scipp/variable/transform.h"
+#include "scipp/core/element/sort.h"
+
+using namespace scipp::core;
+
+namespace scipp::variable {
+
+Variable sort(const VariableConstView &var, const Dim dim, const SortOrder order) {
+  Variable out(var);
+  if (order == SortOrder::Ascending)
+    transform_in_place<std::tuple<span<int64_t>, span<int32_t>, span<double>, span<float>, span<std::string>>>(
```

Comment:
```suggestion
    transform_in_place(
```
and same below. `core::element::sort_*` defines the supported types, no need to give them twice.

Reply:
Removing the mentioned part of this code leaded to error so I left it in the second push.